### PR TITLE
WasmAppHost: block port# 5060 and 5061, for the webserver

### PR DESCRIPTION
**Pull request by kasperk81**
to avoid slipstream attack https://chromestatus.com/feature/5064283639513088

### CODE REVIEW

File: src/mono/wasm/host/WebServerStartup.cs [hunk 1]
```diff
@@ -52,6 +52,18 @@ public static int StartDebugProxy(string devToolsHost)
...
         return generateRandomPort;
+
+        static int GetNextRandomExcept(Range range, params int[] except)
+        {
+            int current = Random.Shared.Next(range.Start.Value, range.End.Value);
+
+            while (Array.IndexOf(except, current) > -1)
+            {
+                current = Random.Shared.Next(range.Start.Value, range.End.Value);
+            }
```
Comment (kasperk81) [inline src/mono/wasm/host/WebServerStartup.cs:63]:
```suggestion
        {
            int current;
            do
            {
                current = Random.Shared.Next(range.Start.Value, range.End.Value);
            } while (Array.IndexOf(except, current) > -1);
```

**Comment by filipnavara** [comment]
You don't need to loop. It's easy to just specify smaller range and +2 to the result if it's >= 5060.

**Comment by kasperk81** [comment]
this is future-proof for additional port blocking

**Comment by filipnavara** [comment]
You can apply the same idea in a generic way. Just subtract the size of the array of the blocked ports from the random range. The loop over the array of blocked ports and increment the random number R by 1 for each array[i] >= R.

**Comment by pavelsavara** [comment]
Could you please ad comment on why we are avoiding the ports ? Also see what browsers consider unsafe
https://github.com/dotnet/runtime/pull/76236#issuecomment-1262346000

**Comment by kasperk81** [comment]
comments added.

> Also see what browsers consider unsafe [#76236 (comment)](https://github.com/dotnet/runtime/pull/76236#issuecomment-1262346000)

combination of those lists in case it is needed:

```c#
            1, // tcpmux
            7, // echo
            9, // discard
            11, // systat
            13, // daytime
            15, // netstat
            17, // qotd
            19, // chargen
            20, // FTP-data
            21, // FTP-control
            22, // SSH
            23, // telnet
            25, // SMTP
            37, // time
            42, // name
            43, // nicname
            53, // domain
            69, // TFTP
            77, // priv-rjs
            79, // finger
            87, // ttylink
            95, // supdup
            101, // hostriame
            102, // iso-tsap
            103, // gppitnp
            104, // acr-nema
            109, // POP2
            110, // POP3
            111, // sunrpc
            113, // auth
            115, // SFTP
            117, // uucp-path
            119, // nntp
            123, // NTP
            135, // loc-srv / epmap
            137, // NetBIOS
            139, // netbios
            143, // IMAP2
            161, // SNMP
            179, // BGP
            389, // LDAP
            427, // SLP (Also used by Apple Filing Protocol)
            465, // SMTP+SSL
            512, // print / exec
            513, // login
            514, // shell
            515, // printer
            526, // tempo
            530, // courier
            531, // Chat
            532, // netnews
            540, // UUCP
            548, // afpovertcp [Apple addition]
            554, // rtsp
            556, // remotefs
            563, // NNTP+SSL
            587, // ESMTP
            601, // syslog-conn
            636, // LDAP+SSL
            989, // ftps-data
            990, // ftps
            993, // IMAP+SSL
            995, // POP3+SSL
            1719, // H323 (RAS)
            1720, // H323 (Q931)
            1723, // H323 (H245)
            2049, // NFS
            3659, // apple-sasl / PasswordServer [Apple addition]
            4045, // lockd
            5060, // SIP
            5061, // SIPS
            6000, // X11
            6566, // SANE
            6665, // Alternate IRC [Apple addition]
            6666, // Alternate IRC [Apple addition]
            6667, // Standard IRC [Apple addition]
            6668, // Alternate IRC [Apple addition]
            6669, // Alternate IRC [Apple addition]
            6697, // IRC+SSL [Apple addition]
            10080, // amanda
            4190, // ManageSieve [Apple addition]
            6679, // Alternate IRC SSL [Apple addition]
```

**Review by pavelsavara — APPROVED** [review]
LGTM after whitespace fix

**Review by radical — APPROVED** [review]

**Pull request merged**